lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,427 @@
+                # values corresponding to percentiles.
+                percentile_bounds[1, indices] = percentiles[index + 1]
+                value_bounds[1, indices] = percentiles_cube[index+1].data[
+                    indices]
+            except IndexError:
+                # Invoked if we have reached the top of the available values.
+                percentile_bounds[1, indices] = percentiles[index]
+                value_bounds[1, indices] = pslice.data[indices]
+
+        with np.errstate(divide='ignore'):
+            numerator = (threshold_cube.data - value_bounds[0])
+            denominator = np.diff(value_bounds, n=1, axis=0)[0]
+            interpolants = numerator/denominator
+            interpolants[denominator == 0] = np.inf
+
+        with np.errstate(invalid='ignore'):
+            probabilities.data, = (percentile_bounds[0] + interpolants *
+                                   np.diff(percentile_bounds, n=1, axis=0))
+        probabilities.data = probabilities.data/100.
+
+        above_top_band = np.isinf(interpolants)
+        below_bottom_band = np.isnan(value_bounds[0])
+        probabilities.data[below_bottom_band] = 0.
+        probabilities.data[above_top_band] = 1.
+
+        return probabilities
+
+    def process(self, threshold_cube):
+        """
+        Slice the percentiles cube over any non-spatial coordinates
+        (realization, time, etc) if present, and call the percentile
+        interpolation method for each resulting cube.
+
+        Args:
+            threshold_cube (iris.cube.Cube):
+                A cube of values, that effectively behave as thresholds, for
+                which it is desired to obtain probability values from a
+                percentiled reference cube.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                A cube of probabilities obtained by interpolating between
+                percentile values at the "threshold" level.
+        """
+        cube_slices = self.percentiles_cube.slices(
+            [self.percentile_coordinate, self.percentiles_cube.coord(axis='y'),
+             self.percentiles_cube.coord(axis='x')])
+
+        if threshold_cube.ndim != 2:
+            msg = ('threshold cube has too many ({} > 2) dimensions - slicing '
+                   'to x-y grid'.format(threshold_cube.ndim))
+            warnings.warn(msg)
+            threshold_cube = next(threshold_cube.slices([
+                threshold_cube.coord(axis='y'),
+                threshold_cube.coord(axis='x')]))
+
+        if threshold_cube.units != self.percentiles_cube.units:
+            threshold_cube.convert_units(self.percentiles_cube.units)
+
+        output_cubes = iris.cube.CubeList()
+        for cube_slice in cube_slices:
+            output_cube = self.percentile_interpolation(threshold_cube,
+                                                        cube_slice)
+            output_cubes.append(output_cube)
+
+        if len(output_cubes) > 1:
```

Comment:
Do you actually need this if statement? Will merge_cube not work on a cubelist with a single cube in?

Reply:
It seems to work, so I've taken this out.